adding a function and editing the error message for the issue #4151

Saty248 opened the pull request:
<!--
Thank you for using Hardhat and taking the time to send a Pull Request!

If you are introducing a new feature, please discuss it in an Issue or with someone from the team before submitting your change.

Please:
 - consider the checklist items below
 - keep the ones that make sense for your PR, and
 - DELETE the items that DON'T make sense for your PR.
-->

- [x ] Because this PR includes a **bug fix**, relevant tests have been included.

---

<!-- Add a description of your PR here -->
i have changed the error message for the issue #4151 and i've also added a function deleteOnReset() which resets the snapshots array.

let me know what other changes I need to make on this issue.

fvictorio commented:
Hey @Saty248, thanks for this. This is a really interesting idea. It won't work for raw `hardhat_reset` calls, so we should still detect those kind of problems, but that could be a separate efforts.

This needs tests though, and I dont' think the code is formatted correctly.

fvictorio commented on packages/hardhat-network-helpers/src/loadFixture.ts:
```diff
@@ -68,3 +68,8 @@ export async function loadFixture<T>(fixture: Fixture<T>): Promise<T> {
 }
+
+// function to clear the snapshots array when hardhat_reset is called..
+export async function deleteOnReset(){
```
This function name is overspecified. I think `clearSnapshots` would be better.

Saty248 commented:
hey @fvictorio,
 i've changed the function name from deleteOnRequest to clearSnapshots and formated the code using lint.Also I've added a test case for the clearsnapshot on reset.ts under test. Do I need to added additional test cases for hardhat reset with a url and block number?
